### PR TITLE
[Merged by Bors] - chore: improve automated fixes in start_port.sh

### PR DESCRIPTION
**Pull request by jcommelin**
fix certain import statements
move "by" to end of line

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: scripts/start_port.sh [hunk 1]
```diff
@@ -62,12 +62,22 @@ git add "$mathlib4_path"
 git commit -m 'Initial file copy from mathport'
 
 sed -i 's/Mathbin\./Mathlib\./g' "$mathlib4_path"
+sed -i '/^import/{s/[.]Smul/.SMul/g; s/[.]Pnat/.PNat/g}' "$mathlib4_path"
+
+# awk script taken from https://github.com/leanprover-community/mathlib4/pull/1523
+awk '{do {{if (match($0, "^  by$") && length(p) < 98) {p=p " by";} else {if (NR!=1) {print p}; p=$0}}} while (getline == 1) if (getline==0) print p}' "$mathlib4_path"
```
Comment (mo271) [inline scripts/start_port.sh:68]:
```suggestion
awk '{do {{if (match($0, "^  by$") && length(p) < 98 && (!(match(p, "^[ \t]*--.*$")))) {p=p " by";} else {if (NR!=1) {print p}; p=$0}}} while (getline == 1) if (getline==0) print p}' "$mathlib4_path"
```
This makes sure that lines with stray comments will not accidentally considered for merging with the `by`.
(Probably not super important..

File: scripts/start_port.sh [hunk 2]
```diff
@@ -62,12 +62,22 @@ git add "$mathlib4_path"
...
 
 git add Mathlib.lean "$mathlib4_path"
-git commit -m 'Mathbin -> Mathlib; add import to Mathlib.lean'
+git commit \
+	-m 'automated fixes' \
+	-m '' \
+	-m 'Mathbin -> Mathlib' \
+	-m 'fix certain import statements' \
+	-m 'move "by" to end of line' \
+	-m 'add import to Mathlib.lean'
```
Comment (mo271) [inline scripts/start_port.sh:80]: Good idea to summarize these

File: scripts/start_port.sh (hunk 1 above)
Comment (mo271) [inline scripts/start_port.sh:68]: In general, I think just running this awk command will give you the replaced file on stdout. Perhaps I'm not understanding something, but probably try if it works at written. 
If you want to replace it, then try either add `-i -inplace` or write it to a temporary fie with `> tmpfile` and then copy it.

**Review by mo271 — APPROVED** [review]

File: scripts/start_port.sh (hunk 1 above)
Comment (jcommelin) [reply scripts/start_port.sh]: thanks, done

**Comment by rwbarton** [comment]
bors merge

**Pull request closed without merging**